test/source_dialogs_test.rb

```diff
@@ -47,6 +73,27 @@
 
       expect(subject.PreprocessISOURL(url)).to eq(converted)
     end
+
+    it "handles iso with part in url and part in path" do
+      converted = "iso:/insta%20ll/Duomenys%20600%20GB/openSUSE-13.2-DVD-x86_64.iso"
+      url = "iso:/Duomenys%20600%20GB?iso=/openSUSE-13.2-DVD-x86_64.iso&url=dir%3A%2Finsta%20ll%2F"
+
+      expect(subject.PreprocessISOURL(url)).to eq(converted)
+    end
+
+    it "handles properly escaped spaces" do
+      converted = "iso:/install/Duomenys%20600%20GB/openSUSE-13.2-DVD-x86_64.iso"
+      url = "iso:///?iso=openSUSE-13.2-DVD-x86_64.iso&url=dir%3A%2Finstall%2FDuomenys%20600%20GB"
+
+      expect(subject.PreprocessISOURL(url)).to eq(converted)
+    end
+
+    it "handles empty iso uri" do
```

Comment:
NP: Could you add a comment why we need to handle this? It's clear from the diff, but few years later it might be quite difficult to find out...